runners/direct-java/src/main/java/org/apache/beam/runners/direct/portable/GroupAlsoByWindowEvaluatorFactory.java

```diff
@@ -138,30 +126,30 @@ public GroupAlsoByWindowEvaluator(
           .getExecutionContext(application, inputBundle.getKey())
           .getStepContext(
               evaluationContext.getStepName(application));
+      // TODO: extract from input PCollection via the graph, including length-prefixing
```

Comment:
are you talking about length prefixing the window coder?

Reply:
Yes - this logic will be shared, but right now we're temporarily placing null and exploding if this is ever used.